### PR TITLE
Add Reflect API

### PR DESCRIPTION
**Pull request by frostney**
## Summary
- Implement the ES2026 §28.1 Reflect namespace object with all 13 static methods: `apply`, `construct`, `defineProperty`, `deleteProperty`, `get`, `getOwnPropertyDescriptor`, `getPrototypeOf`, `has`, `isExtensible`, `ownKeys`, `preventExtensions`, `set`, and `setPrototypeOf`
- Flag-gated via `ggReflect` (included in `DefaultGlobals`), registered in both interpreted and bytecode pipelines
- Key semantic difference from `Object.*`: Reflect methods throw `TypeError` on non-object targets and return booleans instead of throwing on failure for property operations

## Test plan
- [x] 13 test files with 64 tests and 116 assertions covering all methods (`tests/built-ins/Reflect/`)
- [x] Happy paths, TypeError on non-object targets, edge cases (non-configurable, non-extensible, symbol keys)
- [x] Full test suite passes with no regressions
- [x] Clean build from scratch passes
- [x] Formatter passes (`./format.pas --check`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/built-ins/Reflect/getPrototypeOf.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  test("returns class prototype for instances", () => {
+    class Animal {
+      speak() {
+        return "...";
+      }
+    }
+    const a = new Animal();
+    const proto = Reflect.getPrototypeOf(a);
+    expect(proto !== null).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (frostney) [reply tests/built-ins/Reflect/getPrototypeOf.js]: Fixed in bb51124. The test now asserts `Reflect.getPrototypeOf(a)` identity against `Animal.prototype`.
